lib/segment/src/vector_storage/dense/dynamic_mmap_flags.rs

```diff
@@ -216,15 +216,9 @@ impl DynamicMmapFlags {
 
     /// Count number of set flags
     pub fn count_flags(&self) -> usize {
-        let mut ones = self.flags.count_ones();
-
-        // Subtract flags in extra capacity we don't use
-        // They may have been set before shrinking the bitvec again
-        ones -= (self.status.len..self.flags.len())
-            .filter(|&i| self.get(i))
-            .count();
-
-        ones
+        // Take a bitslice of our set length, count ones in it
+        // This uses bit-indexing, returning a new bitslice, extra bits within capacity are not counted
+        self.flags[..self.status.len].count_ones()
```

Comment:
After closer investigation we can just do this.

The backing [`BitSlice`](https://docs.rs/bitvec/latest/bitvec/slice/struct.BitSlice.html) is already smart enough to: 1. index by bits. 2. to just count bits in range, excluding extra bits in capacity.